feat: sticky header and changeable Sign In link

ikrestov opened the pull request:
This adds additional top level header link property className.

![stoplight-menu](https://user-images.githubusercontent.com/1066322/56738572-ccb03000-6764-11e9-81d8-5cb2881bdbeb.gif)

Sign In/Sign Up link text change is a impure hack in Header component. 
IMO, there is no clean way to drive it from configuration.

\cc @moaiandin

lottamus commented on src/components/Header/config.js:
```diff
@@ -38,6 +38,12 @@ export default {
           type: 'array',
           required: false,
         },
+        {
```
I don't think we need to add this to the config? Why not just add the classname directly in the code?

lottamus commented on netlify/settings.yaml:
```diff
@@ -115,6 +115,7 @@ header:
         - arrow-right
       isButton: true
       title: Sign In
+      className: sign-in-link
```
Just add this in the code directly

ikrestov commented:
I have removed extra config entry and changed background to inherit from header (hero), or default to black.

Not sure why netlify build failed, it build fine locally.

lottamus commented on src/components/Header/index.tsx:
```diff
@@ -212,29 +214,79 @@ export interface IHeader {
+    const { header, meta, color } = this.props;
+    const { unpinned } = this.state;
+    const bgClassName = unpinned ? `bg-${color || 'black'}` : '';
+    let headerItems = unpinned ? header.items.slice() : header.items;
```
Looks like its failing here. Just need to safeguard header && header.items before slicing

![image](https://user-images.githubusercontent.com/7423098/56931523-eb386300-6aa5-11e9-8170-305be0bb0389.png)

ikrestov replied:
Thanks. All green now.

lottamus commented:
Great work on this! I think it looks awesome!
